### PR TITLE
location and regulations selections db support

### PR DESCRIPTION
**Pull request by adamsachs**
OSS-side DB support for [PROD-1372](https://ethyca.atlassian.net/browse/PROD-1372) and [PROD-1456 ](https://ethyca.atlassian.net/browse/PROD-1456).

Since this is a paid feature, the API support is in a corresponding [fidesplus PR](https://github.com/ethyca/fidesplus/pull/1269).

### Description Of Changes

DB table and corresponding SQLAlchemy class for persisting state of location and regulation selections. We don't store a lot of data in the DB, at least for now: just a single-row table with two array columns to maintain the active selections. 

I've included some utility class methods (both sync and async versions) for easy getters and setters of the single-record table.

### Code Changes

* [x] DB table migration
* [x] SQLAlchemy class with utility classmethods (sync and async versions)
* [x] test coverage: async tests needed to go in a different subdir to work well with our existing fixtures

### Steps to Confirm

* [x] we've done some pretty solid integration testing with the UI and API in the corresponding fidesplus PR and using the UI branch. we may need to make some API-level tweaks, but the DB layer seems relatively stable and should get us what we need for an initial iteration 👍 

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

[PROD-1372]: https://ethyca.atlassian.net/browse/PROD-1372?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by adamsachs** [comment]
TODO: need to make a readme entry but want to let CI run to completion! 😬

File: src/fides/api/alembic/migrations/versions/4ced99dabebb_add_location_regulation_selections_table.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        ),
+        sa.Column("selected_locations", sa.ARRAY(sa.String()), nullable=False),
+        sa.Column("selected_regulations", sa.ARRAY(sa.String()), nullable=False),
+        sa.Column(
+            "single_row",
+            sa.Boolean(),
+            nullable=False,
+        ),
+        sa.PrimaryKeyConstraint("id"),
+        sa.CheckConstraint("single_row", name="single_row_check"),
```
Comment (TheAndrewJackson) [inline src/fides/api/alembic/migrations/versions/4ced99dabebb_add_location_regulation_selections_table.py:42]: I didn't know it was possible to have a single row constraint. That's cool!

Comment (TheAndrewJackson) [reply src/fides/api/alembic/migrations/versions/4ced99dabebb_add_location_regulation_selections_table.py:42]: ahh, I see. Only one row can have the true boolean value and that value is default to `true`

**Review by TheAndrewJackson — APPROVED** [review]
Looks good!

File: src/fides/api/alembic/migrations/versions/4ced99dabebb_add_location_regulation_selections_table.py (hunk 1 above)
Comment (adamsachs) [reply src/fides/api/alembic/migrations/versions/4ced99dabebb_add_location_regulation_selections_table.py:42]: haha yup, it's a bit hacky but i think this is the best way to do it!

**Pull request merged**